src/test/java/it/gov/pagopa/idpay/transactions/event/CommandsConsumerConfigIntegrationTest.java

```diff
@@ -0,0 +1,190 @@
+    @SpyBean
+    private RewardTransactionRepository rewardTransactionRepository;
+
+    @Test
+    void test() {
+        int validMessages = 100;
+        int notValidMessages = errorUseCases.size();
+        long maxWaitingMs = 30000;
+
+        List<String> commandsPayloads = new ArrayList<>(notValidMessages+validMessages);
+        commandsPayloads.addAll(IntStream.range(0,notValidMessages).mapToObj(i -> errorUseCases.get(i).getFirst().get()).toList());
+        commandsPayloads.addAll(buildValidPayloads(notValidMessages, notValidMessages+validMessages));
+
+        long timeStart=System.currentTimeMillis();
+        commandsPayloads.forEach(cp -> kafkaTestUtilitiesService.publishIntoEmbeddedKafka(topicCommands, null, null, cp));
+        long timePublishingEnd = System.currentTimeMillis();
+
+        waitForLastStorageChange((validMessages/VALID_USE_CASES)*2*NUMBER_TRX);
+
+        long timeEnd=System.currentTimeMillis();
+
+        checkRepositories();
+        checkErrorsPublished(notValidMessages, maxWaitingMs, errorUseCases);
+
+        System.out.printf("""
+                        ************************
+                        Time spent to send %d (%d + %d) messages (from start): %d millis
+                        Time spent to assert db stored count (from previous check): %d millis
+                        ************************
+                        Test Completed in %d millis
+                        ************************
+                        """,
+                commandsPayloads.size(),
+                validMessages,
+                notValidMessages,
+                timePublishingEnd - timeStart,
+                timeEnd - timePublishingEnd,
+                timeEnd - timeStart
+        );
+
+        long timeCommitCheckStart = System.currentTimeMillis();
+        Map<TopicPartition, OffsetAndMetadata> srcCommitOffsets = kafkaTestUtilitiesService.checkCommittedOffsets(topicCommands, groupIdCommands, commandsPayloads.size());
+        long timeCommitCheckEnd = System.currentTimeMillis();
+
+        System.out.printf("""
+                        ************************
+                        Time occurred to check committed offset: %d millis
+                        ************************
+                        Source Topic Committed Offsets: %s
+                        ************************
+                        """,
+                timeCommitCheckEnd - timeCommitCheckStart,
+                srcCommitOffsets
+        );
+
+    }
+
+
+    private long waitForLastStorageChange(int n) {
+        long[] countSaved={0};
+        //noinspection ConstantConditions
+        TestUtils.waitFor(()->(countSaved[0]=rewardTransactionRepository.findAll().count().block()) == n, ()->"Expected %d saved transactions in db, read %d".formatted(n, countSaved[0]), 60, 1000);
+        return countSaved[0];
+    }
+
+    private List<String> buildValidPayloads(int startValue, int messagesNumber) {
```

Comment:
you are calling this parameter "messagesNumber", but this is not the number of elements produced by this method :)

In order to be a real "messagesNumber", you should call the range method as follow:
IntStream.range(startValue, startValue+messagesNumber)

You should rename the parameter, or change the way you are calling this method passing a real "messagesNumber", fixing the range method invoke
`buildValidPayloads(notValidMessages, validMessages)`